### PR TITLE
Add some documentation about how to think about rescheduling jobs

### PR DESCRIPTION
**Pull request by drusellers**
Please leave any and all feedback that can make this page better. I'm happy to go through all the revisions needed. :) This was super helpful to get me to think about things via different approaches.

### CODE REVIEW

File: docs/documentation/quartz-3.x/how-tos/reschedule-jobs.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Reschedule Jobs
+
+A few ways to approach a need to reschudel a job.
```
Comment (lahma) [inline docs/documentation/quartz-3.x/how-tos/reschedule-jobs.md:3]: typo: reschudel

File: docs/documentation/quartz-3.x/how-tos/reschedule-jobs.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+## Manually Retry
+
+When a Quartz job is running, and an unhandled exception escapes the `IJob`, the Quartz system will mark the job in an error state. This would then allow you to reschedule the job using any method that would be work for your system.
```
Comment (lahma) [inline docs/documentation/quartz-3.x/how-tos/reschedule-jobs.md:7]: Maybe should mention the pattern of `try`-`catch` handling exceptions inside the job and then throwing job `JobExecutionException` to notify what should be done. This was more easily enforced in Java version's API as only `JobExecutionException` was allowed to be get out of the job. Exception allows you to signal whether retry should be immediately tried or whether to unschedule all triggers (this job cannot succeed ever).

File: docs/documentation/quartz-3.x/how-tos/reschedule-jobs.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+A few ways to approach a need to reschudel a job.
+
+## Manually Retry
+
+When a Quartz job is running, and an unhandled exception escapes the `IJob`, the Quartz system will mark the job in an error state. This would then allow you to reschedule the job using any method that would be work for your system. 
+
+## Polly Retries
+
+If your job simply needs to retry its work, then you could wrap the job in a [Polly](https://github.com/App-vNext/Polly) policy, and use the policy definitions to retry it.
```
Comment (lahma) [inline docs/documentation/quartz-3.x/how-tos/reschedule-jobs.md:11]: Maybe add a note about long retry loops inside jobs can be potentially harmful as they can exhaust resources if there's thread pool starvation.

File: docs/documentation/quartz-3.x/how-tos/reschedule-jobs.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+## Self-Rescheduling
+
+If your job needs more time, say it needs to wait 5 minutes, the `IJobExecutionContext` has access to the scheduler on it. You could use that to reschudel the job, and let it exit normally.
```
Comment (lahma) [inline docs/documentation/quartz-3.x/how-tos/reschedule-jobs.md:15]: job could also schedule also another job, this is a way to get job-chaining / workflows: https://www.quartz-scheduler.net/documentation/faq.html#how-do-i-chain-job-execution-or-how-do-i-create-a-workflow

File: docs/documentation/quartz-3.x/how-tos/reschedule-jobs.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+```csharp
+public async Task Execute(IJobExecutionContext context)
+{
+    // something happens, that tells you to delay the processing
+    // like getting an HTTP 429 - Too Many requests
+    var oldTrigger = context.Trigger;
+    var newTrigger = TriggerBuilder.Create()
+        .WithIdentity($"{oldTrigger.Key.Name}-retry", oldTrigger.Key.Group)
+        .StartAt(DateTimeOffset.Now + TimeSpan.FromMinutes(5))
```
Comment (lahma) [inline docs/documentation/quartz-3.x/how-tos/reschedule-jobs.md:25]: should always use `.UtcNow` , maybe `DateTimeOffset.UtcNow.AddMinutes(5)` reads more fluently

**Review by lahma — COMMENTED** [review]
Thanks for creating a new documentation section! Added some feedback.

File: docs/documentation/quartz-3.x/how-tos/reschedule-jobs.md (hunk 4 above)
Comment (drusellers) [reply docs/documentation/quartz-3.x/how-tos/reschedule-jobs.md:15]: I think this becomes another how to.

**Comment by drusellers** [comment]
@lahma ok ready for a second pass I believe

**Review by lahma — APPROVED** [review]
Thank you for this, always great to have examples for common scenarios!

**Pull request merged**